### ISSUE TITLE
[Feature] Implement Tower placing

### ISSUE THREAD

**Issue by EinsKatze** [martin-coding/godot-os-war#6]
i guess the title is somewhat clear enough to understand what we need to do.

The player needs to be able to place the Towers all over the map, but should not be able to place it on the enemy path and not on top of another tower.
If you have further suggestions comment them here.

**Comment by mathew-tran** [comment]
do we want tower placement to stick on the grid?

**Comment by mathew-tran** [comment]
Agreed. I think having it stick to grid might feel a bit annoying, also it makes it easier to work with haha.

**Comment by EinsKatze** [comment]
A lot of Features were added in #9 

TODO:
- Prevent Tower Placing on enemy path
- Implement to sell/delete a Tower

(Upgrades will be a different issue, because we need to think about how we want to implement them and how they should behave)

**Comment by mathew-tran** [comment]
I think we should implement this along with upgrading towers and have it in the same UI~

**Comment by EinsKatze** [comment]
Implemented the ability to cancel building in Commit https://github.com/martin-coding/godot-os-war/commit/ab041943e39eca5e04b100ca8fd89d9b2e4f7bc2
